code/_onclick/other_mobs.dm

```diff
@@ -135,6 +134,68 @@
 /mob/living/simple_animal/slime/RestrainedClickOn(atom/A)
 	return
 
+
+/*
+	Drones
+*/
+/mob/living/simple_animal/drone/UnarmedAttack(atom/A)
+	A.attack_drone(src)
+
+/atom/proc/attack_drone(mob/living/simple_animal/drone/user)
+	attack_hand(user) //defaults to attack_hand. Override it when you don't want drones to do same stuff as humans.
+
+/mob/living/simple_animal/slime/RestrainedClickOn(atom/A)
+	return
+
+
+/*
+	True Devil
+*/
+
+/mob/living/carbon/true_devil/UnarmedAttack(atom/A, proximity)
+	A.attack_hand(src)
+
+/*
+	Brain
+*/
+
+/mob/living/brain/UnarmedAttack(atom/A)//Stops runtimes due to attack_animal being the default
+	return
+
+
+/*
+	pAI
+*/
+
+/mob/living/silicon/pai/UnarmedAttack(atom/A)//Stops runtimes due to attack_animal being the default
+	return
+
+
+/*
+	Simple animals
+*/
+
+/mob/living/simple_animal/UnarmedAttack(atom/A, proximity)
+	if(!dextrous)
+		return ..()
+	if(!ismob(A))
+		A.attack_hand(src)
+		update_hand_icons()
+
+
+/*
+	Hostile animals
+*/
+
+/mob/living/simple_animal/hostile/UnarmedAttack(atom/A)
+	target = A
+	if(dextrous && !is_type_in_typecache(A, environment_target_typecache) && !ismob(A))
+		..()
+	else
+		AttackingTarget()

```

Comment:
Moved all those UnarmedAttack() in on_click folder for consistency. Also adds attack_drone() proc, to prevent drones from calling attack_hand() on a given object.